components/style/Cargo.toml

```diff
@@ -30,6 +30,7 @@ euclid = "0.10.1"
 fnv = "1.0"
 heapsize = "0.3.0"
 heapsize_derive = {version = "0.1", optional = true}
+html5ever = {version = "0.10.2", features = ["heap_size", "unstable"]}
```

Comment:
This is a fairly large dependency to add to Stylo, just to use a small enum. And it is also what causes `build-geckolib` to fail because that uses an older (stable channel) Rust compiler that doesn’t know about new-style procedural macros.

Instead I recommend duplicating the definition of the `QuirksMode` enum into `components/style/context.rs`. For consistency, the rest of Servo should be changed to use that enum instead of `html5ever::tree_builder::QuirksMode`. Namely, these files:

```
components/script/dom/node.rs
components/script/dom/document.rs
components/script/dom/bindings/trace.rs
components/script/dom/servoparser/html.rs
```

The latter has a `set_quirks_mode` method that takes a `html5ever::tree_builder::QuirksMode` and will need to do the conversion to `style::context::QuirksMode`.